api-2.2/src/main/java/org/openmrs/module/coreapps/htmlformentry/EncounterDiagnosesElement.java

```diff
@@ -329,8 +331,7 @@ public void handleSubmission(FormEntrySession formEntrySession, HttpServletReque
                     if (diagnosis.getEncounter().getEncounterType() == null) {
                         diagnosis.getEncounter().setEncounterType(formEntrySession.getForm().getEncounterType());
                     }
-
-                    Context.getEncounterService().saveEncounter(diagnosis.getEncounter());
+                    Context.getService(HtmlFormEntryService.class).applyActions(formEntrySession);
```

Comment:
Ideally, we wouldn't be applying the actions directly from a component like this... the element would be creating new actions that would be executed later. 

But, yeah, that ship sailed a long time ago... there are many other examples of where directly modify stuff within a submission element, and this seems better than what we are currently doing... saving the encounter directly within the element.

Reply:
>the element would be creating new actions that would be executed later

@Ruhanga can you look into this?

Reply:
Yep, spiking on a better to handle the encounter diagnoses submission.

Reply:
So what I've done here is avoid what was in the first place causing issues, empty obs groups, then save directly.